### ISSUE TITLE
Missing assertion error when trying to store inf/NaN value in hyperparameter DB

### ISSUE THREAD

**Issue by jkterry1** [optuna/optuna#2850]
So I use Optuna for optimizing deep RL things. For a variety of reasons, deep RL is very unstable and often very numerically unstable and often results in infinities and NaNs with bad hyperparameters (which you find a lot of during hyperparameter optimization). I accidentally tried to store an inf in my database during distributed tuning, except SQL databases generally cannot store infs or NaNs, and that resulted in an absolutely massive error message that was very hard to parse out, and I'm even still somewhat unclear on what specific value I was trying to store that was inf. It seems like a reasonable and useful feature for Optuna to just catch these errors before sending the values to the SQL database and produce a reasonable error message, and I'm fairly surprised Optuna does not already have this. An example error message of this and so on is available in this issue in a downstream package that depends on Optuna:

https://github.com/DLR-RM/rl-baselines3-zoo/issues/146

**Comment by nzw0301** [comment]
Duplicated issue: https://github.com/optuna/optuna/issues/2809
